### DISCUSSION TITLE
Behaviour of labels and addLabes

### DISCUSSION THREAD

**Discussion by jaapcoomans** [renovatebot/renovate#28132] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub Enterprise 3.11.5

### Please tell us more about your question or problem

Based on the documentation I assumed that a matching packageRule with `labels` would override any preceding packageRule that defines `addLabels`.

> The labels array is non-mergeable, meaning if multiple packageRules match then Renovate uses the last value for labels. If you want to add/combine labels, use the addLabels config option, which is mergeable.

In practice though, this is not the behaviour we observe. Our config looks like this (top rules omitted for brevity):
```
"packageRules": [
    {
      "matchUpdateTypes": ["minor"],
      "addLabels": ["minor release"]
    },
    {
      "matchDepTypes": ["test", "action"],
      "labels": ["patch release"]
    }
]
```
Now when a dependency of type "action" gets a "minor" upgrade, the result is that the PR has both the "minor release" and the "patch release" label. Did I misinterpret the documentation, or is this a bug?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by secustor** [comment, marked as answer]
Mergeable refers here to while evaluating different config sources e.g. global, organizational or repo configs as well as packageRules. 

That means if use `addLabels` all values are combined, tough if you use `labels` previous set values of the field `labels` will be overridden. 

What you want is this: 
```json
"packageRules": [
    {
      "matchUpdateTypes": ["minor"],
      "labels": ["minor release"]
    },
    {
      "matchDepTypes": ["test", "action"],
      "labels": ["patch release"]
    }
]
``` 
The result is: 
- all minor updates which do NOT have depType `test` or `action` --> label: "minor release"
- all updates which have depType `test` and `action` --> label: "patch release"

**Reply by jaapcoomans** [reply]
Thanks, this was not exactly clear to me from reading the docs as a not so experienced user.

**Reply by rarkins** [reply]
Docs PRs always welcome if you think you can pay it forward to future users!
